Update spendable balance and ED spec (#6869)
w3f/polkadot-wiki: docs/learn/learn-account-balances.md

Where `free`, `frozen`, and `reserved` are defined above. The `ED` is the
[existential deposit](./learn-accounts.md#existential-deposit-and-reaping).

If an account has any other provider references then the spendable amount is:

```
spendable = free - max(frozen - reserved)
```

### Consumer and Provider References

> In Polkadot and Substrate-based chains, each account must maintain a minimum total balance called the Existential Deposit (ED) to remain alive. An account cannot be reaped from the state while it has a reserved balance, or in general,

Author: ```suggestion
In Polkadot and Substrate-based chains, each account must maintain a minimum free balance called the Existential Deposit (ED) to remain alive (or have another provider see later). An account cannot be reaped from the state while it has a reserved balance, or in general,
```

> In Polkadot and Substrate-based chains, each account must maintain a minimum free balance called the Existential Deposit (ED) to remain alive (or have another provider see later). An account cannot be reaped from the state while it has a reserved balance, or in general,

Author: ```suggestion
In Polkadot and Substrate-based chains, each account must maintain a minimum free balance called the Existential Deposit (ED) to remain alive (or have another provider). An account cannot be reaped from the state while it has a reserved balance, or in general,
```

If an account’s free balance falls below this threshold and has no providers or consumers, it is reaped, meaning all its data is deleted to conserve state space. However, the ED rules are more nuanced than they may appear:

> - Reserved balances do not count as spendable, but they do add a provider reference to the account. (No they don't? otherwise ED would not be untouchable after having more than ED on hold. and ED is untouchable even when having more than ED on hold)

Reviewer: ```suggestion
- Reserved balances do not count as spendable.
```

> - If an account has any provider references (e.g., via staking), it will not be reaped even if its total balance drops below the ED. In such cases, the ED is not untouchable: the total balance can be entirely spent, and the account will remain alive. In other words, the account is "dusted", which happens when its total balance drops below ED. Then the "dust" is deleted, but the account can still be alive, for example, because it has a provider reference.

Author: Actually in practice I have no idea what is the exact spec. I can show tests where account has less than ED in total balance and nothing is dusted and in some other circumstances, it get dusted.. I couldn't find any consistency, so I am not sure what to write here.

```suggestion
- If an account has any provider references (e.g., via staking), it will not be reaped even if its free balance drops below the ED. In such cases, the ED is not untouchable: the total balance can be entirely spent, and the account will remain alive. In other words, the account may be "dusted", which happens ?sometimes? when its free balance drops below ED. Then the "dust" is deleted, but the account can still be alive, for example, because it has a provider reference.
```

> - If an account has any provider references (e.g., via staking), it will not be reaped even if its free balance drops below the ED. In such cases, the ED is not untouchable: the total balance can be entirely spent, and the account will remain alive. In other words, the account may be "dusted", which happens ?sometimes? when its free balance drops below ED. Then the "dust" is deleted, but the account can still be alive, for example, because it has a provider reference.

Author: ```suggestion
- If an account has any provider references (e.g., via staking), it will not be reaped even if its free balance drops below the ED. In such cases, the ED is not untouchable: the free balance can be entirely spent, and the account will remain alive. In other words, the account may be "dusted", which happens ?sometimes? when its free balance drops below ED. Then the "dust" is deleted, but the account can still be alive, for example, because it has a provider reference.
```
